integration/admin-client/src/main/java/org/keycloak/admin/client/CreatedResponseUtil.java

```diff
@@ -39,9 +42,24 @@ public static String getCreatedId(Response response) throws WebApplicationExcept
         URI location = response.getLocation();
         if (!response.getStatusInfo().equals(Response.Status.CREATED)) {
             Response.StatusType statusInfo = response.getStatusInfo();
-            throw new WebApplicationException("Create method returned status " +
-                    statusInfo.getReasonPhrase() + " (Code: " + statusInfo.getStatusCode() + "); " +
-                    "expected status: Created (201)", response);
+            String contentType = response.getHeaderString(HttpHeaders.CONTENT_TYPE);
+            String errorMessage = "Create method returned status " +
+                                  statusInfo.getReasonPhrase() + " (Code: " + statusInfo.getStatusCode() + "); " +
+                                  "expected status: Created (201).";
+            if (MediaType.APPLICATION_JSON.equals(contentType)) {
```

Comment:
This could also potentially be something other than `application/json`, e.g. could include a charset as well `application/json; charset=utf-8`. There is also a `APPLICATION_JSON_TYPE` constant, perhaps the `MediaType` class has a way to compare these as well, without comparing the attributes in the string.

Reply:
@jonkoops Good point

This is backport of already merged community PR. So I've created another PR into `main` : https://github.com/keycloak/keycloak/pull/36558 . Can you review that one please?

Among your point, I've also figured that in some cases, we are returning `error` JSON claim instead of `errorMessage` (admin console is able to handle both though, so we can support both here as well).

Also added some automated testing for this.

Reply:
If we go with #36558 , I can include that one into this PR as well to have it backported both.

Reply:
Agreed, let's merge that together, also an individual backport would be fine

Reply:
@jonkoops Thanks! Merged the original PR into main and added the commit to this PR, so we have the change on single place.